src/kademlia/rpc_manager.cpp

```diff
@@ -300,18 +300,17 @@ bool rpc_manager::incoming(msg const& m, node_id* id
 		, total_milliseconds(now - o->sent()), print_endpoint(m.addr).c_str());
 #endif
 
+	if (m.message.dict_find_string_value("y") == "e") 
+	{
+		// it's an error.
+		o->reply(m);
+		return false;
+	}
+
 	bdecode_node ret_ent = m.message.dict_find_dict("r");
 	if (!ret_ent)
 	{
-		// it may be an error
-		ret_ent = m.message.dict_find("e");
 		o->timeout();
-		if (!ret_ent)
-		{
-			entry e;
```

Comment:
Question is, can we handle a error for a response? I guess no.
I checked code. a error message will be dispatched to rpc_manager::incoming from node::incoming, but then
```
bool rpc_manager::incoming(msg const& m, node_id* id
	, libtorrent::dht_settings const& settings)
{
	INVARIANT_CHECK;

	if (m_destructing) return false;

	// we only deal with replies and errors, not queries
	TORRENT_ASSERT(m.message.dict_find_string_value("y") == "r"
		|| m.message.dict_find_string_value("y") == "e");

	// if we don't have the transaction id in our
	// request list, ignore the packet

	std::string transaction_id = m.message.dict_find_string_value("t");
	if (transaction_id.empty()) return false;
```
since there is no transaction_id, this message will be ignored.
Actually, incoming_error should only be reported for a incoming_request. for response message, the sender don't have a observer/transaction at all, means we couldn't handle it at all.

And even we could, In my mind, DHT is more like a simple RPC, request->response, if response is incorrect, re-request or ignore it.